zuul/connection/github.py

```diff
@@ -345,58 +363,67 @@ def onLoad(self):
     def onStop(self):
         self.unregisterHttpHandler(self.payload_path)
 
-    def _authenticateGithubAPI(self):
-        config = self.connection_config
-
+    def _createGithubClient(self):
         if self.git_host != 'github.com':
             url = 'https://%s/' % self.git_host
             github = github3.GitHubEnterprise(url)
         else:
             github = github3.GitHub()
 
+        return github
+
+    def _authenticateGithubAPI(self):
+        config = self.connection_config
+
         api_token = config.get('api_token')
 
+        integration_id = config.get('integration_id')
+        integration_key = None
+        integration_key_file = config.get('integration_key')
+
+        self._github = self._createGithubClient()
+
         if api_token:
-            github.login(token=api_token)
-        else:
```

Comment:
Why are we removing the else level? If we login with the api_token, don't we want to skip the integration stuff?